LibreSSL 3.8 dropped support for EC_GFp_nist_method()

LibreSSL 4.1 dropped support for EC_METHOD

Index: ext/openssl/ossl_pkey_ec.c
--- ext/openssl/ossl_pkey_ec.c.orig
+++ ext/openssl/ossl_pkey_ec.c
@@ -746,6 +746,7 @@ static VALUE ossl_ec_group_initialize(int argc, VALUE 
 
     switch (rb_scan_args(argc, argv, "13", &arg1, &arg2, &arg3, &arg4)) {
     case 1:
+#if 0
         if (SYMBOL_P(arg1)) {
             const EC_METHOD *method = NULL;
             ID id = SYM2ID(arg1);
@@ -755,7 +756,7 @@ static VALUE ossl_ec_group_initialize(int argc, VALUE 
             } else if (id == s_GFp_mont) {
                 method = EC_GFp_mont_method();
             } else if (id == s_GFp_nist) {
-                method = EC_GFp_nist_method();
+                method = NULL;
 #if !defined(OPENSSL_NO_EC2M)
             } else if (id == s_GF2m_simple) {
                 method = EC_GF2m_simple_method();
@@ -768,7 +769,9 @@ static VALUE ossl_ec_group_initialize(int argc, VALUE 
             } else {
                 ossl_raise(rb_eArgError, "unknown symbol, must be :GFp_simple, :GFp_mont, :GFp_nist or :GF2m_simple");
             }
-        } else if (rb_obj_is_kind_of(arg1, cEC_GROUP)) {
+        } else
+#endif
+        if (rb_obj_is_kind_of(arg1, cEC_GROUP)) {
             const EC_GROUP *arg1_group;
 
             SafeGetECGroup(arg1, arg1_group);
